docs/style-guide.md

```diff
@@ -12,7 +12,7 @@ It's mainly an complement to the [Rust Style Guide](https://pingcap.github.io/st
 
 ## Formatting
 
-- Place all `mod` declaration before any `use`.
+- Place all `mod` declaration before any `use` (except the test mod).
```

Comment:
I don't think we need this statement; it seems to cause more confusion (at least I don't get the point before read the PR description).

```
mod tests {
  ...
}
```

is mod + content, not bare "`mod` declarations".

Reply:
Okay, as we also have a clippy [lint](https://rust-lang.github.io/rust-clippy/master/index.html#/items_after_test_module) to check this. I'll remove it